controllers/neutronapi_controller.go

```diff
@@ -809,6 +815,126 @@ func (r *NeutronAPIReconciler) transportURLCreateOrUpdate(instance *neutronv1bet
 	return transportURL, op, err
 }
 
+func getExternalSecretName(instance *neutronv1beta1.NeutronAPI, serviceName string) string {
+	return fmt.Sprintf("%s-%s-neutron-config", instance.Name, serviceName)
+}
+
+func getMetadataAgentSecretName(instance *neutronv1beta1.NeutronAPI) string {
+	return getExternalSecretName(instance, "ovn-metadata-agent")
+}
+
+func (r *NeutronAPIReconciler) reconcileExternalMetadataAgentSecret(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *neutronv1beta1.NeutronAPI,
+	envVars *map[string]env.Setter,
+) error {
+	sbCluster, err := ovnclient.GetDBClusterByType(ctx, h, instance.Namespace, map[string]string{}, ovnclient.SBDBType)
+	if err != nil {
+		err = r.deleteExternalSecret(ctx, h, instance, getMetadataAgentSecretName(instance))
```

Comment:
Not sure about that. But I think what we should do is to eventually introduce a separate status type to track externalConfigs, so that users can check the status in CRD.